blogposts/2020/announcing-sourcegraph-3.16.md

```diff
@@ -0,0 +1,249 @@
+
+[**📊 Understanding user activity of your instance**](#understanding-user-activity-of-your-instance)<br />
+
+[**ⓘ Improvements to basic code intelligence for several languages**](#improvements-to-basic-code-intelligence-for-several-languages)<br />
+
+[**📝 Changelog**](#changelog)<br />
+Every detail that changed in this release
+
+[**🎖️ Thank you**](#thank-you)<br />
+Sourcegraph couldn't be what it is without the community.
+
+</div>
+
+**Deploy or upgrade:** [Local](https://docs.sourcegraph.com/#quickstart-guide) | [AWS](https://github.com/sourcegraph/deploy-sourcegraph-aws) | [DigitalOcean](https://marketplace.digitalocean.com/apps/sourcegraph?action=deploy&refcode=48dfb3ccb51c) | [Kubernetes cluster](https://github.com/sourcegraph/deploy-sourcegraph)
+
+<div class="alert alert-info">
+  <p>
+    <strong>LEARN MORE:</strong><br />
+    <a href="https://info.sourcegraph.com/online-meetup-may21?utm_source=blog">
+      Online meetup: Sourcegraph 3.16 - Thursday, May 21 at 10AM PDT
+    </a>
+  </p>
+</div>
+
+## Introducing version contexts to search past releases
+
+TODO: video
+
+<!--
+<p class="container">
+  <div style="padding:56.25% 0 0 0;position:relative;">
+    <iframe src="https://www.youtube.com/embed/{VIDEO_ID}" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
+  </div>
+  <p style="text-align: center"><a href="https://youtu.be/{VIDEO_ID}" target="_blank">View on YouTube</a></p>
+</p>
+-->
+
+Many large organizations have old versions of code running in production and need to search across all the code for that release. Sourcegraph 3.16 introduces the ability to create a context consisting of many repositories with specific revisions to be searched. As you explore search results, you might need to navigate your code. Version contexts are respected by code intelligence hovers (go to definition and find references), and allow you to quickly traverse the specific code revisions you care about.
+
+In this release, any non-master branches that are specified in the context will use our non-indexed search. This means that results may be slow for larger deployments. We are working on adding support for indexing additional branches that are specified within the context, and will include this in a future release.
+
+To take advantage of of this feature, add a `versionContexts` block to the experimental features in your site configuration:
+
+```json
+"experimentalFeatures": [
+  "versionContexts": [
+    {
+      "name": "srcgraph 3.16",
+      "revisions": [
+        {
+          "repo": "github.com/sourcegraph/sourcegraph",
+          "ref": "v3.16.0"
+        },
+        {
+          "repo": "github.com/sourcegraph/zoekt",
+          "ref": "b56036a3b745033badc48807bd67e91cd8d73bdf"
+        }
+      ]
+    }
+  ]
+]
+```
+
+## Campaigns
```

Comment:
@sqs or @mrnugget to write initial draft (I am happy to help edit!).

Reply:
```suggestion
## Campaigns

Besides multiple smaller bug fixes and performance improvements, we aimed to make it easier and more straightforward to start using campaigns on your own Sourcegraph instance by providing clearer instructions in the onboarding process.

We also made it easier to setup the [campaigns webhooks](https://docs.sourcegraph.com/user/campaigns/getting_started), which we highly recommend to reduce load on the code host and to improve the speed with which campaigns are updated in Sourcegraph.

The [documentation for campaigns](https://docs.sourcegraph.com/user/campaigns) has also undergone a complete overhaul and now features a video that explains what campaigns can do:
```